spec/controllers/api/v1/project_preferences_controller_spec.rb

```diff
@@ -150,63 +150,85 @@
         }
       }
     end
+    let(:run_update) { post :update_settings, settings_params }
 
     before(:each) do
       default_request user_id: authorized_user.id, scopes: scopes
     end
 
     it "responds with a 200" do
-      put :update_settings, settings_params
+      run_update
       expect(response.status).to eq(200)
     end
 
+    it "responds with the updated resource" do
+      run_update
+      updated_upp = json_response[api_resource_name].first
+      expect(updated_upp).not_to be_empty
+      modified = upp.updated_at.httpdate
+      expect(response.headers).to include('Last-Modified' => modified)
+    end
+
     it "updates the UPP settings attribute" do
-      put :update_settings, settings_params
+      run_update
       found = UserProjectPreference.where(project: project, user: upp.user).first
       expect(found.settings["workflow_id"]).to eq(settings_params[:project_preferences][:settings][:workflow_id].to_s)
     end
 
-    it "cannot update preferences" do
-      sneaky_params =
+    context "with sneaky params" do
+      let(:settings_params) do
         {
           project_preferences: {
             user_id: upp.user_id,
             project_id: project.id,
             preferences: { mail_me_all_the_time: true }
           }
         }
-      put :update_settings, sneaky_params
-      expect(response.status).to eq(422)
-    end
+      end
 
-    it "only updates settings of owned project" do
-      unowned_project = create(:project)
-      new_upp = create(:user_project_preference, project: unowned_project)
+      it "cannot update preferences" do
+        run_update
+        expect(response.status).to eq(422)
+      end
+    end
 
-      nefarious_params =
+    context "with nefarious params" do
+      let(:unowned_project) { create(:project) }
+      let(:new_upp) do
+        create(:user_project_preference, project: unowned_project)
+      end
+      let(:settings_params) do
         {
           project_preferences: {
             user_id: new_upp.user_id,
             project_id: unowned_project.id,
             settings: { workflow_id: 1234 }
           }
         }
-      put :update_settings, nefarious_params
-      expect(response.status).to eq(403)
+      end
+
+      it "only updates settings of owned project" do
+        run_update
+        expect(response.status).to eq(403)
+      end
     end
 
-    it "responds with a 404 if UPP not found" do
-      unused_project = create(:project)
-      silly_params =
+    context "with silly params" do
```

Comment:
With the restructuring you did here, it might be good to make this a little more descriptive.